compiler/rustc_mir_transform/src/dest_prop.rs

```diff
@@ -507,23 +539,6 @@ impl<'a> Conflicts<'a> {
 
     fn record_terminator_conflicts(&mut self, term: &Terminator<'_>) {
         match &term.kind {
-            TerminatorKind::DropAndReplace {
```

Comment:
why don't we need this anymore? Because we run so late this variant doesn't occur? Should probably group the ones that can't happen into a `=> bug!()` arm